src/app/core/services/action-trigger-state.service.ts

```diff
@@ -29,49 +31,61 @@ export class ActionTriggerStateService {
     )
     this.subscriptions.push(
-      this.actionTriggerEventObserver.subscribeToActionTriggerUpdated((actionTriggerUpdatedEvent: ActionTriggerUpdatedEvent) => this.updateActionTrigger(actionTriggerUpdatedEvent))
+      this.actionTriggerEventObserver.subscribeToActionTriggerUpdated((actionTriggerUpdatedEvent: ActionTriggerUpdatedEvent) => this.updateActionTriggerState(actionTriggerUpdatedEvent))
     )
     this.subscriptions.push(
-      this.actionTriggerEventObserver.subscribeToActionTriggerDeleted((actionTriggerDeletedEvent: ActionTriggerDeletedEvent) => this.removeActionTrigger(actionTriggerDeletedEvent))
+      this.actionTriggerEventObserver.subscribeToActionTriggerDeleted((actionTriggerDeletedEvent: ActionTriggerDeletedEvent) => this.removeActionTriggerState(actionTriggerDeletedEvent))
     )
   }
 
-  private addCreatedActionTrigger(actionTriggerCreatedEvent: ActionTriggerCreatedEvent): void {
-    const actionTriggers: ActionTrigger[] = this.actionTriggerSubject.value
-    actionTriggers.push(actionTriggerCreatedEvent.actionTrigger)
-    this.actionTriggerSubject.next(actionTriggers)
-  }
-
-  private updateActionTrigger(actionTriggerUpdatedEvent: ActionTriggerUpdatedEvent): void {
-    const actionTriggers: ActionTrigger[] = this.actionTriggerSubject.value
+  private updateActionTriggerState(actionTriggerUpdatedEvent: ActionTriggerUpdatedEvent): void {
+    const actionTriggers: ActionTrigger<KeyboardTriggerData>[] = this.actionTriggerSubject.value
     const index: number = actionTriggers.findIndex(actionTrigger => actionTrigger.id === actionTriggerUpdatedEvent.actionTrigger.id)
     if (index === -1) {
-      throw new Error(`Updated ActionTrigger does not exist in ActionTrigger state`)
+      this.openDangerSnackBar('Failed to update shortcut')
+      return
     }
-
     actionTriggers[index] = actionTriggerUpdatedEvent.actionTrigger
     this.actionTriggerSubject.next(actionTriggers)
+    this.openSnackBar('Successfully update shortcut')
   }
 
-  private removeActionTrigger(actionTriggerDeletedEvent: ActionTriggerDeletedEvent): void {
-    const actionTriggers: ActionTrigger[] = this.actionTriggerSubject.value
+  private removeActionTriggerState(actionTriggerDeletedEvent: ActionTriggerDeletedEvent): void {
+    const actionTriggers: ActionTrigger<KeyboardTriggerData>[] = this.actionTriggerSubject.value
     const index: number = actionTriggers.findIndex(actionTrigger => actionTrigger.id === actionTriggerDeletedEvent.actionTriggerId)
     if (index === -1) {
       // ActionTrigger is already deleted from the state.
+      this.openDangerSnackBar('Failed to delete shortcut')
       return
     }
     actionTriggers.splice(index, 1)
     this.actionTriggerSubject.next(actionTriggers)
+    this.openSnackBar('Successfully deleted shortcut')
+  }
+
+  private addCreatedActionTriggerState(actionTriggerCreatedEvent: ActionTriggerCreatedEvent): void {
```

Comment:
The final name of this method is `createActionTriggerFromEvent` it's discuss it in @KvelaGorrrrnio comments

Reply:
I'm fine with that suggestion.
I will point out that depending on your naming philosophy then `createActionTrigger(event: Event)` convey the same information as `createActionTriggerFromEvent(...)` since the `fromEvent` is derived from the parameter.
But it comes down to personal preference :)

Reply:
For now, I'm thinking of not changing it and leaving them `createActionTriggerFromEvent`.